### PR TITLE
Add antispam module

### PR DESCRIPTION
**Pull request by ddorgan**

### CODE REVIEW

File: charts/matrix-server/templates/configmap-antiscamspamchecker.yaml [hunk 1]
```diff
@@ -0,0 +1,289 @@
+apiVersion: v1
```
Comment (fgimenez) [inline charts/matrix-server/templates/configmap-antiscamspamchecker.yaml]: I think we should only declare this configmap when antiscam is true, just adding `{{ if .Values.antiscam }}` at the beginning of this file (and `{{end}}` at the end) should be enough

File: charts/matrix-server/templates/statefulset.yaml [hunk 2]
```diff
@@ -38,6 +38,8 @@ spec:
           value: "0"
         - name: GID
           value: "0"
+        - name: PYTHONPATH
```
Comment (fgimenez) [inline charts/matrix-server/templates/statefulset.yaml]: This should be also included only `{{ if .Values.antiscam }}`

File: charts/matrix-server/templates/statefulset.yaml [hunk 3]
```diff
@@ -80,6 +82,10 @@ spec:
               name: matrix-server-log
           - secret:
               name: matrix-server-signing-key
+          - configMap:
```
Comment (fgimenez) [inline charts/matrix-server/templates/statefulset.yaml]: Same as above `{{ if .Values.antiscam }}`

File: charts/matrix-server/values.yaml [hunk 4]
```diff
@@ -5,8 +5,8 @@ port: 8008
 domain: web3.foundation
 
 image:
-  repo: matrixdotorg/synapse
-  tag: v0.99.5.2
+  repo: web3f/synapse
```
Comment (fgimenez) [inline charts/matrix-server/values.yaml]: I think we should not change it here, instead we should set the antiscam values in the community matrix server, here https://github.com/w3f/community-matrix-server/blob/master/helmfile.d/10-matrix.yaml#L13 This way other matrix servers will be unaffected.

File: charts/matrix-server/values.yaml [hunk 5]
```diff
@@ -28,3 +28,5 @@ emailPattern: |-
 
 autojoinRooms:
 - "#arrivals:web3.foundation"
+
+antiscam: true
```
Comment (fgimenez) [inline charts/matrix-server/values.yaml]: Same as above, the default should be false and it should be set to true in the community matrix server chart values, here https://github.com/w3f/community-matrix-server/blob/master/helmfile.d/10-matrix.yaml#L13

**Comment by ddorgan** [comment]
@fgimenez looking good now?

**Pull request merged**